pkg/plugins/types.go

```diff
@@ -0,0 +1,5 @@
+package plugins
+
+type PrometheusAction interface {
+	Run()
```

Comment:
It is recommended to abstract the logic of the underlying lock acquisition in here.